compat/package.json

```diff
@@ -19,7 +19,7 @@
 	"exports": {
 		".": {
 			"types": "./src/index.d.ts",
-			"browser": "./dist/compat.module.js",
+			"browser": "./dist/compat.module.mjs",
 			"umd": "./dist/compat.umd.js",
 			"import": "./dist/compat.mjs",
```

Comment:
we already have .mjs here, correct me if I'm wrong but apart from chromium browsers don't support the .mjs extension yet

Reply:
Browsers don’t care about the extension, they all support `.mjs` on that regard. 


> Jest should be using exports.import rather than exports.browser

Jest is using exports.browser when using the `jest-environment-jsdom` environment, you may or may not agree with that but that’s how they’re doing it 😅 anyway, Preact should be using `.mjs`, or add `"type": "module"` and use `.cjs` extension for CJS file, otherwise it’s hurting their users that use tooling that follow the same logic as Node.js.

Reply:
> it’s hurting their users that use tooling that follow the same logic as Node.js.

It's not, Node uses this correctly as it imports from `import` and not from `browser` if Jest want to import from the browser tag it feels more appropriate to start disregarding the `type: module` stuff and rely on the browser semantics rather than the node ones 😅

Reply:
Any reason not to use `.mjs` though?

Reply:
The thing is we _could_ change the export map to use `.mjs` but we would have to leave the `.module.js` file intact as people could be relying on that. Doing some quick checks and unpkg/skypack do leverage the correct mime when serving from their CDN. As we are not touching the UMD file we can be _quite_ sure we don't have to check if people fill in the correct mime there 😅 CC @developit @marvinhagemeister are you seeing any issues here that I am overlooking?

Reply:
I believe the reason we didn't go with `.mjs` was because it forced webpack into strict module mode which was causing duplicate copies of Preact for codebases with mixed import+require. I believe that problem is now solved via the `module` export field, so we could move exclusively to using `.mjs`.